### PR TITLE
Kraken: fix websocket orderbook processing

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

This pull request addresses several issues and optimizations related to sizing and checksum handling in the application:

1. **Fixed Sizing and Checksum Issue**: The issue with sizing and checksum when subscribed to depths other than 1000 has been resolved. We have made adjustments to the `MaxDepth` field, which is now set as a snapshot deployment instead of being determined by incremental updates.

2. **Resubscribe Issue Fix**: Outbound pair formatting has been updated to adhere to Kraken websocket requirements, resolving the resubscribe issue.

3. **Batch Subscribing and Sleep**: Batch subscribing is no longer limited. Additionally, the unnecessary sleep for subscribing has been removed.

4. **LUNA Pairs Compatibility**: LUNA pairs checksums are failing, this should not hold up merge though.

5. **GST/EUR CRC32 Checksum**: GST/EUR pair, which was previously found to be illiquid and causing a unsub/resub loop, will undergo CRC32 checksum. ie remove len(10) requirement

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [x] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Nice! Just one thing.

Strangely _we_ did have some instances where LUNA pair update checksum was fine, so _we'll_ double check the docs later

File: exchanges/kraken/kraken_websocket.go [hunk 1]
```diff
@@ -857,11 +858,11 @@ func (k *Kraken) wsProcessOrderBook(channelData *WebsocketChannelData, data map[
 
 	askSnapshot, askSnapshotExists := data["as"].([]interface{})
 	bidSnapshot, bidSnapshotExists := data["bs"].([]interface{})
-	if askSnapshotExists || bidSnapshotExists {
-		return k.wsProcessOrderBookPartial(channelData, askSnapshot, bidSnapshot)
+	if !askSnapshotExists && !bidSnapshotExists {
+		return errNoWebsocketOrderbookData
```
Comment (gloriousCode) [inline exchanges/kraken/kraken_websocket.go:862]: Please add the currency pair from channel data

**Review by gloriousCode — CHANGES_REQUESTED** [review]
real quick one

**Review by gloriousCode — APPROVED** [review]
tANANAS!

**Review by thrasher- — APPROVED** [review]
tACKERACK, thanks for expanding the test coverage too 🚀

**Pull request merged**